src/panels/config/zha/zha-groups-dashboard.ts

```diff
@@ -70,7 +70,9 @@ export class ZHAGroupsDashboard extends LitElement {
   }
 
   private async _fetchGroups() {
-    this._groups = (await fetchGroups(this.hass!)).sort(sortZHAGroups);
+    this._groups = this.hass
```

Comment:
You won't have groups now when you are connected and hass is not set yet, so you would need to check if the groups are fetched when hass is set?

Reply:
Is this a question? Sorry, not really following...

Reply:
I guess you added this because your element is connected before `hass` is set? But your solution is to return an empty array, and I don't see you retry it? So you would stay in a broken state?

Reply:
What is the proper solution? I’ll make any changes necessary for the correct fix. I’m still learning the correct ways to handle front end stuff.

Reply:
Check
https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/mixins/subscribe-mixin.ts#L19-L22

https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/mixins/subscribe-mixin.ts#L34-L38

https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/mixins/subscribe-mixin.ts#L46-L50

Replace `this.__unsubs` with a variable that will reflect if you already fetched the groups, or make groups undefined untill you fetched it instead of defaulting to an empty array.

Reply:
Thx, on it

Reply:
how's this?